tests/cypress/support/commands.js

```diff
@@ -740,3 +721,63 @@ Cypress.Commands.add('checkOlLayer', (args = null) => {
         })
     })
 })
+
+Cypress.Commands.add('checkCesiumLayer', (args = null) => {
+    const { layers, visibleLayers } = parseCheckLayerArgs(args)
+
+    cy.window().its('cesiumViewer.scene.imageryLayers').as('cesiumImageryLayers')
+    cy.get('@cesiumImageryLayers').then((imageryLayers) => {
+        const layerIds = layers.map((l) => l.id)
+        const cesiumLayerIds = layerIds
+            .map((id, index) => `[${index}]:${imageryLayers.get(index)?.id}`)
+            .join(',')
+        cy.log(
+            `Check if layers [${layerIds.join(',')}] are set correctly in Cesium [${cesiumLayerIds}]`
+        )
+    })
+
+    // Here in order to have the cy.log() working as expected we need to do the loop outside the
+    // cy.get('@cesiumImageryLayers')
+    visibleLayers.forEach((layer, index) => {
+        cy.get('@cesiumImageryLayers').then((imageryLayers) => {
+            cy.log(
+                `Check that visible layer ${layer.id} at zIndex ${index} is set correctly in Cesium`
+            )
+            const cesiumlayer = imageryLayers.get(index)
+            expect(cesiumlayer, `[${layer.id}] layer at index ${index} not found`).not.to.be.null
+            expect(cesiumlayer, `[${layer.id}] layer at index ${index} not found`).not.to.be
+                .undefined
+            expect(cesiumlayer.show, `[${layer.id}] layer.visible`).to.be.equal(layer.visible)
+            expect(cesiumlayer.alpha, `[${layer.id}] layer.opacity`).to.be.equal(layer.opacity)
+            // The rendered flag is set asynchronously therefore we need to do some retry here
+            cy.waitUntil(() => cesiumlayer.ready, {
+                description: `[${layer.id}] waitUntil layer.ready`,
+                errorMsg: `[${layer.id}] layer.ready is not true`,
+            })
+        })
+    })
+})
+
+Cypress.Commands.add('clickOlMapAtCoordinate', (coordinate, options = {}) => {
+    const { doubleClick = false } = options
+    cy.readWindowValue('map').then((map) => {
+        const pixel = map.getPixelFromCoordinate(coordinate).map(Math.floor)
+        if (pixel.some((value) => value < 0)) {
+            cy.log(
+                'Error, coordinate',
+                coordinate,
+                ' are outside current map viewport (',
+                pixel,
+                '), cannot click on these coordinate'
+            )
+            cy.wrap(false).should('be.true')
+        } else {
+            cy.log('Clicking at coordinate', coordinate, 'on pixel', pixel, 'with options', options)
+            if (doubleClick) {
+                cy.get('[data-cy="ol-map"]').dblclick(pixel[0], pixel[1])
+            } else {
+                cy.get('[data-cy="ol-map"]').click(pixel[0], pixel[1])
+            }
```

Comment:
Here the `:visible` selector should be added. Also this is kind of dangerous, as the computed pixel might be out of the ol map viewport !
We should move the map to the center of the coordinate before by modifying the url center, then you can simply do a click('center')